src/contractor/contractor.cpp

```diff
@@ -40,6 +41,16 @@ template <> struct hash<std::pair<OSMNodeID, OSMNodeID>>
         return static_cast<uint64_t>(k.first) ^ (static_cast<uint64_t>(k.second) << 12);
     }
 };
+
+template <> struct hash<std::tuple<OSMNodeID, OSMNodeID, OSMNodeID>>
+{
+    std::size_t operator()(const std::tuple<OSMNodeID, OSMNodeID, OSMNodeID> &k) const
+    {
+        return (static_cast<uint64_t>(std::get<0>(k)) << 24) ^
+               (static_cast<uint64_t>(std::get<1>(k)) << 12) ^
+               (static_cast<uint64_t>(std::get<2>(k)));
```

Comment:
Instead of an own hash functio you should consider using `boost:hash_combine`:

http://www.boost.org/doc/libs/1_55_0/doc/html/hash/combine.html